src/ORM/Association.php

```diff
@@ -749,6 +749,15 @@ public function attachTo(Query $query, array $options = []): void
                     $this->getName()
                 ));
             }
+            if (
+                !empty($options['matching']) &&
+                $this->_strategy === static::STRATEGY_JOIN &&
+                !empty($dummy->getContain())
```

Comment:
there should be no method usage here in empty() calls, it can be removed without issues here.

Reply:
Can you rephrase that?

Reply:
`&& $dummy->getContain()`
suffices

The cloaking method check is not needed and an anti pattern in these cases of existing property, key or method.

Reply:
That's fine, just couldn't understand the sentence so wanted to double check.